docs/src/content/docs/core/with-plugins.mdx

```diff
@@ -0,0 +1,24 @@
+---
+title: withPlugins
+description: Allows to extend hooks with custom parameters. Advanced.
+---
+
+`withPlugins` allows to extend hooks with custom parameters. Advanced.
```

Comment:
This `Advanced.` at the end of the sentence feels abrupt, maybe this could be replaced with a note box saying something like  "For advance use-cases only, we recommend using `defineUtility`"

Reply:
Agreed! I'll update it